src/uproot/writing/_cascade.py

```diff
@@ -1760,7 +1761,7 @@ def add_directory(self, sink, name, initial_directory_bytes, uuid, flush=True):
             )
             next_key = subdirectory_key.copy_to(self._data.next_location)
         self._data.add_key(next_key, incremental=True)
```

Comment:
Assign it before adding it, just in case `add_key` ever changes to mean "copy the `next_key`, then add it."

```suggestion
        next_key._location = self._data.next_location
        self._data.add_key(next_key, incremental=True)
```